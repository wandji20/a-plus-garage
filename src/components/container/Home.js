import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import AddTrackButton from '../presentation/AddTrackButton';
import Cars from './Cars';
import getLogInDetails from '../../redux/actions/logInAction';
import { getToken } from '../../helpers/session';

const Home = (props) => {
  const {
    loggedIn, fetchUserInfo, cars,
  } = props;

  const carIds = cars.map((car) => car.id);

  useEffect(() => {
    const token = getToken('TOKEN');
    if (!loggedIn && token !== '') {
      fetchUserInfo(token);
    }
  }, [loggedIn]);

  useEffect(() => {
  }, [cars]);

  return (
    <div className="container remove-padding d-flex justify-content-center flex-wrap">
      {
        (loggedIn && cars.length > 0)
          ? <Cars carIds={carIds} />
          : <AddTrackButton />
      }
    </div>
  );
};

const mapStateToProps = (state) => ({
  loggedIn: state.userReducer.loggedIn,
  cars: state.userReducer.cars,
});

const mapDispatchToProps = (dispatch) => ({
  fetchUserInfo: (data) => {
    dispatch(getLogInDetails(data));
  },
});

Home.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  fetchUserInfo: PropTypes.func.isRequired,
  cars: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
