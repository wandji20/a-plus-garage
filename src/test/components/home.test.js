import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import Home from '../../components/container/Home';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
