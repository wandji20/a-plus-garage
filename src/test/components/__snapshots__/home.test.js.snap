// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container remove-padding d-flex justify-content-center flex-wrap"
>
  <a
    href="/new_car"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
        "marginTop": "200px",
      }
    }
  >
    <button
      className="btn btn-info"
      type="button"
    >
      Track Car
    </button>
  </a>
</div>
`;
